codegen/src/generator/mod.rs

```diff
@@ -62,7 +65,15 @@ where P: GenerateProtocol {
         }}
         ",
         methods = protocol_generator.generate_methods(service),
-        service_name = &service.metadata.service_abbreviation,
+        service_name = match &service.metadata.service_abbreviation {
+            &Some(ref service_abbreviation) => service_abbreviation.as_str(),
+            &None => {
+                match service.metadata.endpoint_prefix {
```

Comment:
I wonder if we should just switch to endpointprefix instead of service abbreviation for everything.  I think every service has that and may simplify this code.  Gotta check to see what other impacts that may have.